lib/XMLHttpRequest.js

```diff
@@ -391,7 +452,61 @@ function XMLHttpRequest(opts) {
     }
 
     // Load files off the local filesystem (file://)
+    // or data from Data URI (data:)
     if (local) {
+      if (datauri) try {
+        // Triming from original url string since 
+        var data = url.href.substr(5, settings.url.length);
+
+        // separator between header and actual data
+        var parts = data.split(",", 2);
+
+        if (parts.length < 2) throw "Invalid URL";
+
+        var base64 = false;
+        // check if header part has base64
+        var data_headers = parts[0].split(";");
+        
+        // ignore first element since it's MIME type
+        for (var i = 1; i < data_headers.length; ++i) {
+          if (data_headers[i].toLowerCase() === "base64") {
+            base64 = true;
+            break;
+          }
+        }
+
+        var responseData, inputData = decodeURIComponent(parts[1]);
+
+        if (base64) {
+          // remove any ASCII whitespaces
+          inputData = inputData.replace(/(\s|\t|\r|\n|\v|\f)+/g, "");
+          // check padding amount
+          let padding = inputData.match(/=*$/)[0];
+          if (padding.length + (inputData.length - padding.length) % 4 > 4) throw "invalid padding";
+          inputData = inputData.substr(0, inputData.length - padding.length);
+          responseData = Buffer.from(inputData, "base64");
+          if (responseData.toString("base64").replace(/=+$/, "") !== inputData) throw "malformed base64 encoding";
+        }
+        else {
+          responseData = Buffer.from(inputData);
+        }
```

Comment:
Would it be more clear to extract this to a separate function? `toBufferFromDataUri` or something in those lines. I realize that is definitely not the trend in this project, but there's no reason to keep adding to the madness.

Reply:
sure, let me separate it